### PR TITLE
[OpenXLA] Migrate to pull XLA from OpenXLA

### PR DESCRIPTION
**Pull request by ManfeiBai**

### CODE REVIEW

File: third_party/xla_client/BUILD [hunk 1]
```diff
@@ -77,51 +70,44 @@ cc_library(
         ":unique",
         ":util",
         ":xla_util",
-        ":xrt_local_service",
-        ":xrt_session",
-        ":xrt_session_cache",
-        "@org_tensorflow//tensorflow:grpc++",
```
Comment (ManfeiBai) [inline third_party/xla_client/BUILD:83]: cc_library(
    name = "grpc++",
    visibility = ["//visibility:public"],
    deps = select({
        ":linux_s390x": ["@com_github_grpc_grpc//:grpc++_unsecure"],
        "//conditions:default": ["@com_github_grpc_grpc//:grpc++"],
    }),
)

Comment (ManfeiBai) [reply third_party/xla_client/BUILD:83]: workspace(name = "myproj")
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
http_archive(
    name = "com_github_grpc_grpc",
    urls = [
        "https://github.com/grpc/grpc/archive/635693ce624f3b3a89e5a764f0664958ef08b2b9.tar.gz",
    ],
    strip_prefix = "grpc-635693ce624f3b3a89e5a764f0664958ef08b2b9",
)
load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
grpc_deps()
load("@com_github_grpc_grpc//bazel:grpc_extra_deps.bzl", "grpc_extra_deps")
grpc_extra_deps()

File: third_party/xla_client/BUILD [hunk 2]
```diff
@@ -77,51 +70,44 @@ cc_library(
...
+        "@com_github_grpc_grpc//:grpc++", # maybe use "@xla/xla:grpc_lib" according to xla/workspace0.bzl
         "@org_tensorflow//tensorflow/cc:client_session",
         "@org_tensorflow//tensorflow/cc:scope",
```
Comment (ManfeiBai) [inline third_party/xla_client/BUILD:75]: cc_library_with_android_deps(
    name = "scope",
    srcs = [
        "framework/scope.cc",
        "framework/scope_internal.h",
    ],
    hdrs = ["framework/scope.h"],
    android_deps = ["//tensorflow/core:portable_tensorflow_lib"],
    common_deps = [
        ":ops",
    ],
    deps = [
        "//tensorflow/core:core_cpu",
        "//tensorflow/core:framework",
        "//tensorflow/core:lib",
        "//tensorflow/core:protos_all_cc",
        "@com_google_absl//absl/strings",
    ],
)

File: third_party/xla_client/BUILD [hunk 3]
```diff
@@ -77,51 +70,44 @@ cc_library(
...
         "@org_tensorflow//tensorflow/compiler/jit:xla_cpu_device",
```
Comment (ManfeiBai) [inline third_party/xla_client/BUILD:76]: cc_library(
    name = "xla_cpu_device",
    srcs = ["xla_cpu_device.cc"],
    visibility = [":friends"],
    deps = [
        ":common",
        ":flags",
        ":jit_compilation_passes",
        ":xla_device",
        ":xla_kernel_creator",  # buildcleaner: keep
        "@com_google_absl//absl/memory",
        "//tensorflow/compiler/jit/kernels:xla_ops",
        "//tensorflow/compiler/tf2xla:layout_util",
        "//tensorflow/compiler/tf2xla:xla_compiler",
        "//tensorflow/compiler/tf2xla:xla_op_registry",
        "//tensorflow/compiler/tf2xla/kernels:xla_ops",
        "//tensorflow/core:core_cpu_internal",
        "//tensorflow/core:lib",
    ] + if_libtpu(
        if_false = [
            "//tensorflow/compiler/xla/service:cpu_plugin",  # buildcleaner: keep
        ],
        if_true = [],
    ),
    alwayslink = 1,
)

File: third_party/xla_client/BUILD [hunk 4]
```diff
@@ -77,51 +70,44 @@ cc_library(
...
+        "@xla/xla/pjrt:pjrt_c_api_client",
+        "@xla/xla/rpc:grpc_stub",
+        "@xla/xla/service:cpu_plugin",
+        "@xla/xla/service:platform_util",
+        "@xla/xla:statusor",
+        "@xla/xla:xla_data_proto_cc",
+        "@xla/xla/service:hlo_proto_cc",
+        "@xla/xla/hlo/ir:hlo",
+        "@xla/xla/service/spmd:spmd_partitioner",
         "@org_tensorflow//tensorflow/core:core_cpu",
```
Comment (ManfeiBai) [inline third_party/xla_client/BUILD:111]: alias(
    name = "core_cpu",
    actual = "//tensorflow/core/common_runtime:core_cpu",
    visibility = ["//visibility:public"],
)

Comment (ManfeiBai) [reply third_party/xla_client/BUILD:111]: tf_cuda_library(
    name = "core_cpu",
    hdrs = [
        "composite_device.h",
        "device.h",
        "device_factory.h",
        "device_mgr.h",
        "function.h",
        "function_optimization_registry.h",
        "gradients.h",
        "graph_constructor.h",
        "graph_def_builder_util.h",
        "optimization_registry.h",
        "shape_refiner.h",
        "//tensorflow/core/graph:core_cpu_headers",
        "//tensorflow/core/public:session.h",
        "//tensorflow/core/public:session_options.h",
    ],
    visibility = ["//visibility:public"],
    deps = [
        ":core_cpu_internal",
    ],
)

Comment (ManfeiBai) [reply third_party/xla_client/BUILD:111]: tf_cc_test_mkl(
    name = "mkl_related_tests",
    size = "small",
    srcs = [
        "//tensorflow/core/util:mkl_util_test_srcs",
    ],
    linkstatic = 1,
    deps = [
        ":core",
        ":core_cpu",
        ":core_cpu_internal",

Comment (ManfeiBai) [reply third_party/xla_client/BUILD:111]: # ":core_cpu" - exports the public non-test headers for: graph/: Support for graphs made up of ops common_runtime/: Common code for execution of graphs public/: Public APIs for running graphs

# ":core" - The code for ":core_cpu" plus a GPU runtime

File: .bazelrc [hunk 5]
```diff
@@ -54,7 +54,7 @@ build:cuda --repo_env TF_NEED_CUDA=1
 # "compute" means we emit both cubin and PTX, which is larger but also forward compatible to future GPU generations.
 build:cuda --crosstool_top=@local_config_cuda//crosstool:toolchain
 build:cuda --@local_config_cuda//:enable_cuda
-build:cuda --@org_tensorflow//tensorflow/compiler/xla/python:enable_gpu=true
```
Comment (ManfeiBai) [inline .bazelrc:57]: config_setting(
    name = "link_gpu_plugin",
    define_values = {"xla_python_enable_gpu": "true"},
)

bool_flag(
    name = "enable_gpu",
    build_setting_default = True,
)

config_setting(
    name = "gpu_enabled",
    flag_values = {
        ":enable_gpu": "True",
    },
)

File: third_party/PyTorchXLA.bzl [hunk 6]
```diff
@@ -0,0 +1,104 @@
+# copy from "@xla//tensorflow:tensorflow.bzl",
+def ptxla_cc_shared_object(
+        name,
+        srcs = [],
+        deps = [],
+        data = [],
+        linkopts = lrt_if_needed(),
```
Comment (ManfeiBai) [inline third_party/PyTorchXLA.bzl:7]: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tensorflow.bzl#:~:text=%23%20Linux%20systems%20may,%7D)

**Comment by ManfeiBai** [comment]
tried in a newer PR: https://github.com/pytorch/xla/pull/4878
